File: umbrella-insurance-core/src/main/java/com/umbrella/insurance/core/models/entities/Period.java
package com.umbrella.insurance.core.models.entities;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

@Getter
@Setter
@Entity
@Table(name = "periods", schema = "public")
public class Period {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "period_id", nullable = false)
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "period_type_id")
    private PeriodType periodType;

    @NotNull
    @Column(name = "period_number", nullable = false)
    private Long periodNumber;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "period_status_id")
    private PeriodStatus periodStatus;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "game_id")
    private Game game;

}

File: umbrella-insurance-react-ui/src/models/periods/v1/Period.ts
import { Game } from "../../games/v1/Game";
import { PeriodStatus } from "../periodStatuses/v1/PeriodStatus";
import { PeriodType } from "../periodTypes/v1/PeriodType";

export class Period {
    id?: number;
    game?: Game;
    periodType?: PeriodType;
    periodNumber?: number;
    periodStatus?:PeriodStatus;

    constructor(obj?:any) {
        if(obj) {
            if(obj?.id) {
                this.id = obj.id;
            }
            if(obj?.game) {
                this.game = new Game(obj.game);
            }
            if(obj?.periodType) {
                this.periodType = new PeriodType(obj.periodType);
            }
            if(obj?.periodNumber) {
                this.periodNumber = obj.periodNumber;
            }
            if(obj?.periodStatus) {
                this.periodStatus = new PeriodStatus(obj.periodStatus);
            }
        }
    }
}